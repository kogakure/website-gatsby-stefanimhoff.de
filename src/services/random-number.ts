export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
