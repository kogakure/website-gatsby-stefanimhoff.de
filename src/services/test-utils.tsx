import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
