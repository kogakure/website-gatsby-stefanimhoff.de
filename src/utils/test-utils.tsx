import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
