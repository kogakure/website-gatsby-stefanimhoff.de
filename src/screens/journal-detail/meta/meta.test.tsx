import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../../services/test-utils';

import { Meta } from '.';

const data = {
  date: '2019-10-16T18:00:00',
  readingTime: '2 min read',
};

describe('Journal Detail', () => {
  describe('Meta', () => {
    test('renders correctly', () => {
      const { container } = render(<Meta {...data} />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<Meta {...data} />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
