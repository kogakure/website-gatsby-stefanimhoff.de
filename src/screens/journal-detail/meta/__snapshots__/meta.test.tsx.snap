// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journal Detail Meta renders correctly 1`] = `
.c4 {
  cursor: pointer;
}

.c2 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c2:hover,
.c2:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c0 {
  grid-column: 2 / -2;
  grid-row: 2;
}

.c1 {
  line-height: 1;
  margin-bottom: clamp( 0.694rem,1.49vw,0.833rem);
}

.c3 {
  line-height: 1.2;
}

.c5 {
  font-style: italic;
  line-height: 1;
}

@media (min-width:768px) {
  .c0 {
    grid-column: -6 / -3;
  }
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <em>
      By
    </em>
     
    <a
      block="0"
      class="c2"
      href="/about/"
    >
      Stefan Imhoff
    </a>
  </div>
  <div
    class="c3"
  >
    <time
      aria-label="Toggle date format"
      class="c4"
      datetime="2019-10-16T18:00:00"
      role="button"
      tabindex="0"
    >
      October 16th, 2019
    </time>
  </div>
  <div
    class="c5"
  >
    2 min read
  </div>
</header>
`;
